### PR TITLE
MINOR: Move OffsetCheckpointFile to storage module

### PR DESCRIPTION
**Pull request by mimaison**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/log/LogCleanerManager.scala [hunk 1]
```diff
@@ -376,13 +377,13 @@ private[log] class LogCleanerManager(val logDirs: Seq[File],
...
   def updateCheckpoints(dataDir: File,
-                        partitionToUpdateOrAdd: Option[(TopicPartition, Long)] = None,
+                        partitionToUpdateOrAdd: Option[(TopicPartition, JLong)] = None,
                         partitionToRemove: Option[TopicPartition] = None): Unit = {
     inLock(lock) {
       val checkpoint = checkpoints(dataDir)
       if (checkpoint != null) {
         try {
-          val currentCheckpoint = checkpoint.read().filter { case (tp, _) => logs.keys.contains(tp) }.toMap
+          val currentCheckpoint = checkpoint.read().asScala.filter { case (tp, _) => logs.keys.contains(tp) }.toMap
```
Comment (mimaison) [inline core/src/main/scala/kafka/log/LogCleanerManager.scala:386]: It's unfortunate we have to convert the map into Scala here and back to Java below but keeping it in Java is really a pain.

Here is what I had to do to make it work:
```scala
val currentCheckpoint: util.Map[TopicPartition, JLong] = checkpoint.read()
    .entrySet.stream
    .filter(entry => logs.keys.contains(entry.getKey))
    .collect(Collectors.toMap(
        (entry: util.Map.Entry[TopicPartition, JLong]) => entry.getKey,
        (entry: util.Map.Entry[TopicPartition, JLong]) => entry.getValue))
// remove the partition offset if any
var updatedCheckpoint = partitionToRemove match {
  case Some(topicPartition) =>
    currentCheckpoint.remove(topicPartition)
    currentCheckpoint
  case None => currentCheckpoint
}
// update or add the partition offset if any
updatedCheckpoint = partitionToUpdateOrAdd match {
  case Some(updatedOffset) =>
    updatedCheckpoint.put(updatedOffset._1, updatedOffset._2)
    updatedCheckpoint
  case None => updatedCheckpoint
}

checkpoint.write(updatedCheckpoint)
```

File: core/src/main/scala/kafka/log/LogCleanerManager.scala [hunk 2]
```diff
@@ -448,14 +449,16 @@ private[log] class LogCleanerManager(val logDirs: Seq[File],
   /**
    * Truncate the checkpointed offset for the given partition if its checkpointed offset is larger than the given offset
    */
-  def maybeTruncateCheckpoint(dataDir: File, topicPartition: TopicPartition, offset: Long): Unit = {
+  def maybeTruncateCheckpoint(dataDir: File, topicPartition: TopicPartition, offset: JLong): Unit = {
     inLock(lock) {
       if (logs.get(topicPartition).config.compact) {
         val checkpoint = checkpoints(dataDir)
         if (checkpoint != null) {
```
Comment (chia7712) [inline core/src/main/scala/kafka/log/LogCleanerManager.scala:456]: this is unrelated to this PR, but the scala map will throw `NoSuchElementException` when there is no mapping value from the given key. Hence, the null check is redundant.

File: core/src/main/scala/kafka/log/LogCleanerManager.scala [hunk 3]
```diff
@@ -448,14 +449,16 @@ private[log] class LogCleanerManager(val logDirs: Seq[File],
...
           val existing = checkpoint.read()
-          if (existing.getOrElse(topicPartition, 0L) > offset)
-            checkpoint.write(mutable.Map() ++= existing += topicPartition -> offset)
+          if (existing.getOrDefault(topicPartition, 0L) > offset) {
+            existing.put(topicPartition, offset)
```
Comment (chia7712) [inline core/src/main/scala/kafka/log/LogCleanerManager.scala:459]: could you please add comments to `OffsetCheckpointFile#read` to remind the returned map should be mutable.

**Review by chia7712 — COMMENTED** [review]
@mimaison thanks for this nice patch.

File: core/src/main/scala/kafka/log/LogCleanerManager.scala (hunk 2 above)
Comment (mimaison) [reply core/src/main/scala/kafka/log/LogCleanerManager.scala:456]: There's a couple of places with similar code. I'd prefer changing these in a separate PR.

File: core/src/main/scala/kafka/log/LogCleanerManager.scala (hunk 3 above)
Comment (mimaison) [reply core/src/main/scala/kafka/log/LogCleanerManager.scala:459]: That's a good point! I've added a comment

**Review by chia7712 — APPROVED** [review]
LGTM

**Comment by mimaison** [comment]
None of the test failures are related, merging to trunk

**Pull request merged**